app/lib/actions/katello/content_view_environment/reassign_objects.rb

```diff
@@ -5,7 +5,12 @@ class ReassignObjects < Actions::Base
         def plan(content_view_environment, options)
           concurrence do
             content_view_environment.hosts.each do |host|
-              plan_action(Host::Reassign, host, options[:system_content_view_id], options[:system_environment_id])
+              content_facet_attributes = host.content_facet
+              if content_facet_attributes.multi_content_view_environment?
+                content_facet_attributes.content_view_environments -= [content_view_environment]
```

Comment:
note to self for testing: Need to make sure this calls `#content_view_environments=` so that the priority is properly updated.